### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by madejejej**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,33 +29,81 @@ To deploy these nodes, download the jar from the releases tab on github
...
-### Castle Profiler Node
-This node tags the AM login page with the Castle JS to collect information about the event.
+### Before you begin
 
-#### Castle Profiler Node Configuration
-* **App ID** - Castle App ID.
-* **Profiler URI** - Castle client side CDN link.
+1. Create a [Castle account](https://dashboard.castle.io/signup/new)
+2. Once you're signed in, go the the [Dashboard Settings page](https://dashboard.castle.io/settings/general) to configure the Castle Service with the APP ID and API Secret. Please not your Castle environment - you should use the "Sandbox" environment only for non-production use.
+3. We highly suggest you host our JS script yourself, otherwise it might get blocked by ad-blockers.
```
Comment (jstoup111) [inline README.md:36]: Maybe a link to NPM (is that valid in this case) or link to documentation on how to host the JS in our docs.

File: README.md [hunk 2]
```diff
@@ -24,33 +29,81 @@ To deploy these nodes, download the jar from the releases tab on github
...
+
+#### Castle Risk Node Configuration
+
+* **Event** - The Castle Event type.
+* **Status** - The Castle Event status.
+* **Mail Attribute** - The ForgeRock email attribute.
+
+### Castle Filter Node
+
+This node makes a request the Castle's bot detection [Filter API](https://castle.io/filter-api/) to retrieve a policy decision about an action.
```
[the remark this answers is not in the document]
Comment (jstoup111) [reply README.md:87]: Do we actually want to link our reference docs here?

File: README.md [hunk 3]
```diff
@@ -24,33 +29,81 @@ To deploy these nodes, download the jar from the releases tab on github
...
+
+#### Castle Filter Node Configuration
+
+* **Event** - The Castle Event type.
+* **Status** - The Castle Event status.
+* **Mail Attribute** - The ForgeRock email attribute.
+
+### Castle Log Node
+
+This node makes a request the Castle's Log API. This is an async API that provides visibility into the actions your users take.
```
Comment (jstoup111) [inline README.md:97]: Do we want to link to our reference docs?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (madejejej) [reply README.md]: Added a link

File: README.md [hunk 4]
```diff
@@ -24,33 +29,81 @@ To deploy these nodes, download the jar from the releases tab on github
...
 
-### Castle Profiler Node
-This node tags the AM login page with the Castle JS to collect information about the event.
+### Before you begin
 
-#### Castle Profiler Node Configuration
-* **App ID** - Castle App ID.
-* **Profiler URI** - Castle client side CDN link.
+1. Create a [Castle account](https://dashboard.castle.io/signup/new)
+2. Once you're signed in, go the the [Dashboard Settings page](https://dashboard.castle.io/settings/general) to configure the Castle Service with the APP ID and API Secret. Please not your Castle environment - you should use the "Sandbox" environment only for non-production use.
```
[the remark this answers is not in the document]
Comment (madejejej) [reply README.md]: ah, typo

File: README.md (hunk 1 above)
Comment (madejejej) [reply README.md]: I added a link to our NPM package, but we'll have to update the README to make it clear how to host it yourself.

**Pull request merged**